lldbinit.py

```diff
@@ -9,16 +9,174 @@
+    exprMethods = [sorbetType.GetMemberFunctionAtIndex(i) for i in range(sorbetType.GetNumberOfMemberFunctions())]
+    toStringMethod = next((m for m in exprMethods if m.GetName() == "toString"), None)
+    if toStringMethod is None:
+        if error is not None:
+            error.SetErrorString("Object of type %s doesn't have a toString method" % sorbetType)
+        return None
+    # Try to retrieve GS and CFG from the current context
+    gs = get_variable_in_frames(frame, ["sorbet::core::GlobalState", "sorbet::core::Context"])
+    if gs is None:
+        if error is not None:
+            error.SetErrorString("Couldn't find a GS instance in the current frame")
+        return None
+    cfg = None
+    if toStringMethod.GetNumberOfArguments() == 2:
+        cfg = get_variable_in_frames(frame, "sorbet::cfg::CFG")
+        if cfg is None:
+            if error is not None:
+                error.SetErrorString("Couldn't find a CFG instance in the current frame")
+            return None
+    globalStateAccess = "%s%s" % (format_argument(gs), ".state" if "sorbet::core::Context" in gs.GetTypeName() else "")
+    command = sorbetObj.get_expr_path()
+    cmd = "(%s).toString(%s)" % (command, globalStateAccess)
+    if toStringMethod.GetNumberOfArguments() == 2:
+        cmd = "(%s).toString(%s, %s)" % (command, globalStateAccess, format_argument(cfg))
+    finalExpr = frame.EvaluateExpression(cmd)
+    if not finalExpr.IsValid():
+        if error is not None:
+            error.SetErrorString("No result was returned from expression evaluation")
+        return None
+    return finalExpr
+
+def cmd_dump_sorbet(debugger, command, exe_ctx, result, internal_dict):
+    error = lldb.SBError()
+    error.SetErrorToGenericError()
+    frame = exe_ctx.GetFrame()
+    # Value that we are supposed to dump and retrieve its type to find a toString method
+    expr = frame.EvaluateExpression(command)
+    finalExpr = sorbet_obj_toString(expr, error)
+    if finalExpr is not None:
+        result.Print("R: %s\n" % finalExpr)
+    else:
+        result.Print("no result\n")
+        result.SetError(error)
+
+def format_sorbet_core_NameRef(valobj, internal_dict, options):
+    id = valobj.GetChildMemberWithName("_id").GetValueAsSigned(-1)
+    lastKnownID = valobj.GetTarget().FindFirstGlobalVariable("DEBUG_NAME_LAST_WELL_KNOWN_NAME")
+    if id <= 0:
+        return "Default ID"
+    elif lastKnownID is not None and lastKnownID.IsValid() and id < lastKnownID.GetValueAsSigned(0):
+        # Keep bound check in sync with NAME_LAST_WELL_KNOWN_NAME from autogenerated file Names_gen.h
+        name = valobj.EvaluateExpression("::sorbet::core::Names::DEBUG_NAMES_LOOKUP[%d]" % id)
+        name.SetFormat(lldb.eFormatCString)
+        return name.GetValue()
+    else:
+        # Try to get the toString version of it
+        stringified = sorbet_obj_toString(valobj)
+        return ("Dynamic ID %d" % id) if stringified is None else stringified.GetSummary()
```

Comment:
Why would we not always get the toString verison? Shouldn't this code work in both cases?

Reply:
It's possible the script won't be able to find or operate on a valid GS or CFG object so this is the last resort fallback.